RELEASE.md

```diff
@@ -0,0 +1,3 @@
+Release type: minor
+
+Add StrawberryAnnotated, a container type to preserve `typing.Annotated` data when converting types to StrawberryTypes
```

Comment:
```suggestion
This release adds support for `typing.Annotated` so that is preserved  when converting types to StrawberryTypes
```
I would rephrase a little to emphasize the functionality we are adding rather than what changed internally :)

Reply:
Sounds good, thank you